src/jl_uv.c

```diff
@@ -177,94 +81,21 @@ DLLEXPORT void jl_uv_shutdownCallback(uv_shutdown_t *req, int status)
     free(req);
 }
 
-DLLEXPORT void jl_uv_return_spawn(uv_process_t *p, int64_t exit_status, int term_signal)
-{
-    JULIA_CB(return_spawn,p->data,2,CB_INT64,exit_status,CB_INT32,term_signal);
-}
-
-DLLEXPORT void jl_uv_readcb(uv_stream_t *handle, ssize_t nread, const uv_buf_t *buf)
-{
-    JULIA_CB(readcb,handle->data,3,CB_INT,nread,CB_PTR,(buf->base),CB_UINT,buf->len);
-}
-
-DLLEXPORT void jl_uv_alloc_buf(uv_handle_t *handle, size_t suggested_size, uv_buf_t *buf)
-{
-    if (handle->data) {
-        jl_value_t *ret = JULIA_CB(alloc_buf,handle->data,1,CB_UINT,suggested_size);
-        JL_GC_PUSH1(&ret);
-        // TODO: jl_fieldref allocates boxes here. should avoid that.
-        assert(jl_is_tuple(ret) && jl_nfields(ret)==2 && jl_is_pointer(jl_fieldref(ret,0)));
-        buf->base = (char*)jl_unbox_voidpointer(jl_fieldref(ret,0));
-#ifdef _P64
-        assert(jl_is_uint64(jl_fieldref(ret,1)));
-        buf->len = jl_unbox_uint64(jl_fieldref(ret,1));
-#else
-        assert(jl_is_uint32(jl_fieldref(ret,1)));
-        buf->len = jl_unbox_uint32(jl_fieldref(ret,1));
-#endif
-        JL_GC_POP();
-    }
-    else {
-        buf->len = 0;
-    }
-}
-
-DLLEXPORT void jl_uv_connectcb(uv_connect_t *connect, int status)
-{
-    JULIA_CB(connectcb,connect->handle->data,1,CB_INT32,status);
-    free(connect);
-}
-
-DLLEXPORT void jl_uv_connectioncb(uv_stream_t *stream, int status)
-{
-    JULIA_CB(connectioncb,stream->data,1,CB_INT32,status);
-}
-
-DLLEXPORT void jl_uv_getaddrinfocb(uv_getaddrinfo_t *req,int status, struct addrinfo *addr)
-{
-    JULIA_CB(getaddrinfo,req->data,2,CB_PTR,addr,CB_INT32,status);
-    free(req);
-}
-
-DLLEXPORT void jl_uv_asynccb(uv_handle_t *handle)
-{
-    JULIA_CB(asynccb,handle->data,0);
-}
-
-DLLEXPORT void jl_uv_timercb(uv_handle_t *handle)
-{
-    JULIA_CB(timercb,handle->data,0);
-}
-
-DLLEXPORT void jl_uv_pollcb(uv_poll_t *handle, int status, int events)
-{
-    JULIA_CB(pollcb,handle->data,2,CB_INT32,status,CB_INT32,events);
-}
-
-DLLEXPORT void jl_uv_fspollcb(uv_fs_poll_t *handle, int status, const uv_stat_t *prev, const uv_stat_t *curr)
-{
-    JULIA_CB(fspollcb,handle->data,3,CB_INT32,status,CB_PTR,prev,CB_PTR,curr);
-}
+// getters and setters
+DLLEXPORT void *jl_uv_process_data(uv_process_t *p) { return p->data; }
+DLLEXPORT void *jl_uv_buf_base(const uv_buf_t *buf) { return buf->base; }
```

Comment:
We should be able to do this in julia since we have the correct calling convention now.